libraries/node-core-library/src/FileSystem.ts

```diff
@@ -911,6 +933,100 @@ export class FileSystem {
     });
   }
 
+  /**
+   * Copies a single file from one location to one or more other locations.
+   * By default, the file at the destination is overwritten if it already exists.
+   *
+   * @remarks
+   * The `copyFileToManyAsync()` API cannot be used to copy folders.  It copies at most one file.
+   *
+   * The implementation is based on `createReadStream()` and `createWriteStream()` from the
+   * `fs-extra` package.
+   */
+  public static async copyFileToManyAsync(options: IFileSystemCopyFileToManyOptions): Promise<void> {
+    options = {
+      ...COPY_FILE_DEFAULT_OPTIONS,
+      ...options
+    };
+
+    if (FileSystem.getStatistics(options.sourcePath).isDirectory()) {
+      throw new Error(
+        'The specified path refers to a folder; this operation expects a file path:\n' + options.sourcePath
+      );
+    }
+
+    await FileSystem._wrapExceptionAsync(async () => {
+      // See flags documentation: https://nodejs.org/api/fs.html#fs_file_system_flags
+      const writeFlags: string[] = [];
+      switch (options.alreadyExistsBehavior) {
+        case AlreadyExistsBehavior.Error:
+        case AlreadyExistsBehavior.Ignore:
+          writeFlags.push('wx');
+          break;
+        case AlreadyExistsBehavior.Overwrite:
+        default:
+          writeFlags.push('w');
+      }
+      const flags: string = writeFlags.join();
+
+      const createPipePromise: (sourceStream: fs.ReadStream, destinationPath: string) => Promise<void> = (
+        sourceStream: fs.ReadStream,
+        destinationPath: string
+      ) => {
+        return new Promise((resolve: () => void, reject: (error: Error) => void) => {
+          const destinationStream: fs.WriteStream = fs.createWriteStream(destinationPath);
+          const streamsToDestroy: fs.WriteStream[] = [destinationStream];
+          sourceStream.on('error', (e: Error) => {
+            for (const streamToDestroy of streamsToDestroy) {
+              streamToDestroy.destroy();
+            }
+            reject(e);
+          });
+          sourceStream
+            .pipe(destinationStream)
+            .on('close', () => {
+              resolve();
+            })
+            .on('error', async (e: Error) => {
```

Comment:
This seems incorrect.  Do Node.js streams support promises?  If not, then if the `Promise` rejects there will be no catch handler, and the process will get terminated.